app-misc/flirc: initial pull request

github-k8n opened the pull request:
Allows you to pair any remote control with your computer or media center.
https://flirc.tv/

Closes: https://bugs.gentoo.org/582156

monsieurp commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,53 @@
+
+src_prepare () {
+	use qt5 && sed -i -e 's/^Version=.*$/Version=1.0/' "${S}"/usr/share/applications/Flirc.desktop
+	eapply_user
```
Move this line above.

monsieurp commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,53 @@
+		dev-qt/qtcore:5 )"
+
+src_prepare () {
+	use qt5 && sed -i -e 's/^Version=.*$/Version=1.0/' "${S}"/usr/share/applications/Flirc.desktop
```
```
if use qt5; then
    sed -i -e 's/^Version=.*$/Version=1.0/' "${S}"/usr/share/applications/Flirc.desktop || die
fi
```

monsieurp commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,53 @@
+
+inherit eutils unpacker
+
+DESCRIPTION="Allows you to pair any remote control with your computer or media center."
```
Remove the dot please.

Zlogene commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+EAPI=6
+
+inherit eutils unpacker
```
I do not see how `eutils` serves in your ebuild :)

Zlogene commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+
+LICENSE="all-rights-reserved"
+SLOT="0"
+KEYWORDS="amd64"
```
`KEYWORDS` are wrong here. This should be `KEYWORDS="-* ~amd64`

github-k8n replied:
gaah... was just looking at the email and missed that comment ,,, will fix ...

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+S="${WORKDIR}"
+
+DEPEND=""
+RDEPEND="virtual/libusb
```
`virtual/libusb` has two incompatible slots. Given this is a binary package, you need to be specific ;-).

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+src_prepare () {
+	eapply_user
+	if use qt5 ; then
+		sed -i -e 's/^Version=.*$/Version=1.0/' "${S}"/usr/share/applications/Flirc.desktop || die
```
Did you report that upstream? If not, please do.

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+}
+
+src_install () {
+	insinto /lib/udev/rules.d
```
Please use `udev.eclass` for this.

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+	insinto /lib/udev/rules.d
+	newins etc/udev/rules.d/99-flirc.rules 51-flirc.rules
+	doman usr/share/doc/flirc/flirc_util.1
+	exeinto /opt/bin
```
There is no need to use `/opt` here. It's a common misunderstanding but `/opt` is meant for packages that do not adhere to `/usr` structure and e.g. need data files below executable, or are otherwise strongly separate.

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,55 @@
+		exeinto /opt/bin
+		doexe usr/bin/Flirc
+		insinto /usr/share/pixmaps
+		doins usr/share/pixmaps/Flirc.png
```
`doicon` from `desktop.eclass`, and `domenu` below.

github-k8n commented:
hopefully fixed everything and reported upstream (.desktop file issue)

mgorny commented:
Almost there. Just one more issue I see:

```
 * Messages for package app-misc/flirc-3.9.12:
 * Log file: /var/log/portage/app-misc:flirc-3.9.12:20180405-192643.log

 * QA Notice: Files built without respecting CFLAGS have been detected
 *  Please include the following list of files in your report:
 * /usr/bin/Flirc
 * /usr/bin/flirc_util
```

You're looking for QA_PREBUILT.

github-k8n commented:
Hmm.. i can't see anything obvious that might cause the QA checks to crash? .... trying with specifying the files individually now in the QA_PREBUILT variable instead of using wildcard..

mgorny commented:
Nah, it wasn't that. Probably just hit a bad moment on the server or while I was upgrading scripts.

github-k8n commented:
Ok, let me know if you prefer the previous version
QA_PREBUILT="/usr/bin/*"
over the current one specifying the individual files (which i just created because of the crashed QA bot ..)

github-k8n commented:
3.10.0 out which fixed the reported issue with the .desktop file.
Added ebuild for 3.10.0, removed src_prepare stage for that ebuild, reverted QA_PREBUILT back to wildcard

mgorny commented on app-misc/flirc/flirc-3.10.0.ebuild:
```diff
@@ -0,0 +1,45 @@
+
+EAPI=6
+
+inherit unpacker udev desktop
```
Sort, please.

mgorny commented on app-misc/flirc/flirc-3.9.12.ebuild:
```diff
@@ -0,0 +1,52 @@
+# Copyright 1999-2018 Gentoo Foundation
```
Just the newest version, please.

github-k8n commented:
Updated to 3.11.5

mgorny commented:
```
>>> Emerging (1 of 1) app-misc/flirc-3.11.5::gentoo
>>> Downloading 'https://packagecloud.io/Flirc/repo/packages/ubuntu/zesty/flirc_3.11.5_amd64.deb/download.deb'
--2018-04-20 08:35:38--  https://packagecloud.io/Flirc/repo/packages/ubuntu/zesty/flirc_3.11.5_amd64.deb/download.deb
Resolving packagecloud.io (packagecloud.io)... 50.97.198.58, 2607:f0d0:2101:270::2
Connecting to packagecloud.io (packagecloud.io)|50.97.198.58|:443... connected.
HTTP request sent, awaiting response... 404 Not Found
2018-04-20 08:35:39 ERROR 404: Not Found.

!!! Couldn't download 'flirc-3.11.5.deb'. Aborting.
 * Fetch failed for 'app-misc/flirc-3.11.5', Log file:
 *  '/var/log/portage/app-misc:flirc-3.11.5:20180420-063538.log'
```

They don't like us packagers.

github-k8n commented:
Ha, 3.11.6 was released and strangely all older packages were removed...
Will update the ebuild tonight ...

mgorny commented:
I hope they aren't planning to keep removing old versions instantly because that's going to be a maintenance nightmare. I can tell you straight ahead this package can't ever go stable if they do that.

github-k8n commented:
build for 3.11.6 which also adds arm as supported platform (had to change the local filename for the downloaded files to include arch...)

mgorny commented:
Thanks!
